Gems/Atom/Tools/AtomToolsFramework/Code/Source/Viewport/ModularViewportCameraController.cpp

```diff
@@ -254,18 +254,24 @@ namespace AtomToolsFramework
         }
         else if (m_cameraMode == CameraMode::Animation)
         {
+            AZ_Assert(m_cameraAnimation.has_value(), "CameraAnimation is not set when in CameraMode::Animation");
+
             const auto smootherStepFn = [](const float t)
             {
                 return t * t * t * (t * (t * 6.0f - 15.0f) + 10.0f);
             };
 
-            m_cameraAnimation.m_time = AZ::GetClamp(
-                m_cameraAnimation.m_time +
-                    (event.m_deltaTime.count() / ModularViewportCameraControllerRequests::InterpolateToTransformDuration),
-                0.0f,
-                1.0f);
+            if (m_cameraAnimation->m_duration == 0.0f)
```

Comment:
I'm always a bit wary about exact equality tests with floats.

Reply:
I did consider this, but as it's only guarding against a divide by zero, I think the exact check is okay here? 🤔 If you think very small values could still cause a problem I can change it but I explicitly thought in this case it was correct/safe

Reply:
If it's just to avoid a DBZ I guess it's fine. I did just briefly wonder if -0.0f was considered equal to 0.0f but did a quick check and apparently it is.